pvlib/shading.py

```diff
@@ -553,3 +553,109 @@ def shaded_fraction1d(
     )
 
     return np.clip(t_asterisk, 0, 1)
+
+
+def martinez_shade_factor(shaded_fraction, N_shaded_blocks, N_total_blocks):
+    r"""
+    A shading correction factor for the direct and circumsolar incident
+    irradiance of non-monolithic Silicon
+    modules and arrays with an arbitrary number of bypass diodes.
+
+    .. versionadded:: 0.11.0
+
+    Parameters
+    ----------
+    shaded_fraction : numeric
+        Surface shaded fraction. Unitless.
+    shaded_blocks : numeric
+        Number of blocks affected by the shadow. Unitless integer.
+        If a floating point number is provided, it will be rounded up.
+    total_blocks : numeric
```

Comment:
These parameter names don't match the function definition.  I suggest we look through other pvlib functions to see if we have a convention for how to name "number of X" parameters.  `pvlib.snow.dc_loss_nrel` has a `num_strings` parameter, for example.

Reply:
Changed param names to `shaded_blocks` and `total_blocks`!

If you prefer that prefix, I don't have an strong opinion on either. It seems there is no convention used here.

Examples:

```
samples_per_window : int
        Number of data points in each window

modules_per_string: int, default 1
        Number of modules per string in the array.

strings: int, default 1
        Number of parallel strings in the array.

cells_in_series : integer
        The number of cells connected in series.

ivcurve_pnts : int, optional
        Number of points in the desired IV curve. If not specified or 0, no
        points on the IV curves will be produced.

num_strings: int
        The number of parallel-connected strings along a row slant height.

snow_events : array-like
        Number of snowfall events each month. Referred to as N in [1]_. [-]

numthreads : int, optional, default 4
        Number of threads to use if how == 'numba'.

npoints : int, default 100
        Number of points for integration. [unitless]

n_pvrows: int, default 3
        Number of PV rows to consider in the PV array

max_rows : int, default 10
        Maximum number of rows to consider in front and behind the current row.

decimals : int, optional
        number of decimal places to which voltage is rounded to remove
        duplicated points. If not specified, no rounding is done.
```